src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -95,13 +97,38 @@ private static ThreeState GetObsoleteContextState(Symbol symbol, bool forceCompl
 
         internal static ObsoleteDiagnosticKind GetObsoleteDiagnosticKind(Symbol symbol, Symbol containingMember, bool forceComplete = false)
         {
+            if (symbol is NamedTypeSymbol namedTypeSymbol && isExperimentalSymbol(namedTypeSymbol))
+            {
+                // Skip for System.Diagnostics.CodeAnalysis.ExperimentalAttribute to mitigate cycles
+                return ObsoleteDiagnosticKind.NotObsolete;
+            }
+
+            if (symbol is MethodSymbol { MethodKind: MethodKind.Constructor } method && isExperimentalSymbol(method.ContainingType))
+            {
+                // Skip for constructors of System.Diagnostics.CodeAnalysis.ExperimentalAttribute to mitigate cycles
+                return ObsoleteDiagnosticKind.NotObsolete;
+            }
+
+            if (symbol.ContainingModule.ObsoleteKind is ObsoleteAttributeKind.Experimental
+                || symbol.ContainingAssembly.ObsoleteKind is ObsoleteAttributeKind.Experimental)
```

Comment:
Why do we do this here? Couldn't we check containing member only when this `symbol`'s obsolete kind is `None`?

Reply:
You're right, it's better to put there. That said, I don't think one should use both attributes (Obsolete and Experimental) :-P